### PR TITLE
Accept a single-axis mesh (NetKet-style meshes)

### PR DESCRIPTION
**Pull request by PhilipVinc**
(This is a spinoff from getting jaxmg 1.0 to work in netket https://github.com/netket/netket/pull/2265 . FYI: All code was 90% written by claude, but I steered it heavily to write what I wanted, clean it up and make it more understable and follow a design I think it's reasonable)

Jax documentation explicitly state (see https://docs.jax.dev/en/latest/parallel.html ) that `None` in a `PartitionSpec` means that dimension is not sharded, and that "trailing `None` placeholders can be omitted, so that P('X', None) == P('X')".
This is heavily used by netket, for example, which follows this convention heavily in all its code.

However jaxmg requires that this extra dimension be explicitly specified in the mesh. Due to how shard map works, a non trivial amount of code is required to be able to use jamg when all arrays/matrices are specified with a rank-1 mesh.
While I can put this code in netket, I think that jaxmg itself is the better place to support rank-1 meshes and basically treat meshes like (Nr) -> (Nr, 1).

This PR essentially implements this. 

Basically what this PR does is that we call cuda from the context of 'use_abstract_mesh' so that we swap the abstract mesh to always be rank 2 . We do this by decorating those calls with 'use_abstract_mesh_decorator'.

There's a fair bit of logic change in the 'validate_2d_matrix_specs' to support the new setups. Basically the new things we support are:
 - P('S') , remapped to P('S', 1)
 - P('S', None) remapped to P(S, 1)
 - P(None, S) remapped to P(1, S)

What do you think?
I tested the PR on 2 H100 nodes and it passed tests.

### CODE REVIEW

**Comment by JacobTutt** [comment]
Looks great ! Have added a few GPU tests and updated the docs in places (and caught a minor existing issue in the RHS sharding). Jenkins tests have been run and are now passing

**Comment by PhilipVinc** [comment]
Thanks for having a look @JacobTutt ! 
Let me know if you want me to do anything else

**Comment by therooler** [comment]
Looks good to me. One thing.

Right now all examples use a single grid:
```python
mesh = jax.make_mesh((num_processes,), ("pr",))
```
I think we should keep 2/3 examples with the 2D mesh. I propose Cholesky and SYEVD on 2D and LU solve on 1D grid.

We should also communicate to users that a 2D grid will be faster, due to increased parallelizability.

**Comment by PhilipVinc** [comment]
(Is a 2d grid faster even in a very large matrix? it would be nice to see a benchmark to maybe automatically reshard in netket...)

**Comment by therooler** [comment]
Yes, we have some preliminary results showing that the 2D grid can be significantly faster. 

We're in the process of benchmarking the solvers on 64 H200s where we will compare a 8x8 vs 64x1 grid.

Once these results are in, we'll update the paper with the new benchmarks and add them to the docs.
